internal/annotators/http/parser.go

```diff
@@ -0,0 +1,115 @@
+package http
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+func RemoveExtraSpaces(s string) string {
+	return strings.Join(strings.Fields(s), " ")
+}
+
+func HTTPParser(r *http.Request) string {
+
+	//Signature Inputs extraction
+	var rgx = regexp.MustCompile(`\((.*?)\)`)
+	fmt.Printf("r: %v\n", r)
+	signatureInput := r.Header.Get("Signature-Input")
+	fmt.Printf("signatureInput: %v\n", signatureInput)
+	rs := rgx.FindStringSubmatch(signatureInput)
+	signatureInputList := strings.Split(rs[1], " ")
+	fmt.Printf("Signature-Input list: %v\n", signatureInputList)
+
+	signatureInputFields := make(map[string][]string)
+	for i := 0; i < len(signatureInputList); i += 1 {
+		field := signatureInputList[i]
+		//remove double quotes from the field to access it directly in the header map
+		signatureInputList[i] = field[1 : len(field)-1]
+		signatureInputFields[signatureInputList[i]] = []string{}
+	}
+
```

Comment:
You can collapse these loops into one

```
for _, field := range signatureInputList {
	//remove double quotes from the field to access it directly in the header map
	key := field[1 : len(field)-1]
	if key[0:1] == "@" {
		switch key {
		case "@method":
			signatureInputFields[key] = []string{r.Method}
		...
		}
         } else {
		fieldValues := r.Header.Values(key)
               ...
         }
}
```

Also replace string literals like `@method` with constants local to this package

Reply:
Fixed in ef52a4d46b344e24703744e982ac2fc17aed75c4